pkg/controller/atlasdeployment/searchindex.go

```diff
@@ -77,15 +76,15 @@ func (sr *searchIndexReconciler) Reconcile(spec *akov2.SearchIndex, previous *st
 		}
 	}
 
-	return sr.reconcileInternal(stateInAKO, stateInAtlas)
+	return sr.reconcileInternal(name, stateInAKO, stateInAtlas)
 }
 
-func (sr *searchIndexReconciler) reconcileInternal(stateInAKO, stateInAtlas *searchindex.SearchIndex) workflow.Result {
+func (sr *searchIndexReconciler) reconcileInternal(idxName string, stateInAKO, stateInAtlas *searchindex.SearchIndex) workflow.Result {
```

Comment:
suggestion: `idxName` implies many things. The `idx` prefix most likely stands for "index" but is not obvious to the reader. Let's spell this out to `searchIndexName` which makes it very explicit.